validateImages.js

```diff
@@ -2,57 +2,69 @@ const fs = require('fs');
-      console.error(`\nInvalid file type: ${file}`);
-      console.error(`Allowed extensions: ${allowedExtensions.join(', ')}`);
+function validateImages(dataFolderPath) {
+  fs.readdir(dataFolderPath, (err, files) => {
+    if (err) {
+      console.error('Error reading files:', err);
       process.exit(1);
     }
 
-    const filePath = join(dataFolderPath, file);
+    for (let file in files) {
+      if (excludedFiles.includes(files[file])) {
+        continue
+      }
+      console.log(`running validation check on ${files[file]}...`);
 
-    // check if image is lesser than file max size
-    const fileSize = fs.statSync(filePath).size;
-    if (fileSize > maxFileSize) {
-      console.error(`\nFile size exceeds the limit: ${file}`);
-      console.error(`File size: ${fileSize} bytes`);
-      process.exit(1);
-    }
+      // check if image is svg or png
+      const extension = files[file].slice(-4);
+      if (!allowedExtensions.includes(extension)) {
+        console.error(`\nInvalid file type: ${files[file]}`);
+        console.error(`Allowed extensions: ${allowedExtensions.join(', ')}`);
+        // process.exit(1);
+      }
 
-    try {
-      const dimensions = sizeOf(filePath);
-      const width = dimensions.width;
-      const height = dimensions.height;
+      const filePath = join(dataFolderPath, files[file]);
 
-      if (width !== height) {
-        console.error(`\nImage is not square: ${file}`);
-        console.error(`Width: ${width}, Height: ${height}`);
+      // check if image is lesser than file max size
+      const fileSize = fs.statSync(filePath).size;
+      if (fileSize > maxFileSize) {
+        console.error(`\nFile size exceeds the limit: ${files[file]}`);
+        console.error(`File size: ${fileSize} bytes`);
         process.exit(1);
       }
 
-      if ((width < minSize || width > maxSize) && extension !== '.svg') {
-        console.error(`\nImage size is not within the allowed range: ${file}`);
-        console.error(`Width: ${width}, Height: ${height}`);
-        process.exit(1);
+      try {
+        const dimensions = sizeOf(filePath);
+        const width = dimensions.width;
+        const height = dimensions.height;
+
+        if (width !== height) {
+          console.error(`\nImage is not square: ${files[file]}`);
+          console.error(`Width: ${width}, Height: ${height}`);
+          process.exit(1);
+        }
+
+        if ((width < minSize || width > maxSize) && extension !== '.svg') {
+          console.error(`\nImage size is not within the allowed range: ${files[file]}`);
+          console.error(`Width: ${width}, Height: ${height}`);
+          process.exit(1);
+        }
+      } catch (error) {
+        console.error(`Error reading image: ${files[file]}`, error);
       }
-    } catch (error) {
-      console.error(`Error reading image: ${file}`, error);
     }
   });
-});
+}
+
+validateImages(tokensPath);
+validateImages(blockchainsPath);
```

Comment:
For `validateImages`, you might want to return a boolean `hasError`, so that if at least 1 of the `validateImages` function executions returns true, you can `process.exit(1)` to indicate there is an error (clearer indication to the dev that there are errors that need to be fixed).